### PR TITLE
Evaluation for PreHeader properties

### LINKED ISSUE #422: [150 SigmaUSD] Evaluation for `PreHeader` properties
from #418
For each `PreHeader` property implement evaluation in `eval::spreheader` module.
See `sbox`, `scoll` modules for examples.

Change `PreHeader::parent_id` type to `BlockId`.
Change `PreHeader::votes` type to `Votes`.

### PR DESCRIPTION
**Pull request by techraed**
Close #422

### CODE REVIEW

**Comment by greenhat** [comment]
Please resolve the merge conflicts (after #433 merge).

**Comment by SabaunT** [comment]
> Please resolve the merge conflicts (after #433 merge).

done

File: ergotree-interpreter/src/eval/spreheader.rs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        // calling for PreHeader property on Context obj
+        let expr: Expr = PropertyCall {
+            obj: Box::new(Expr::Context),
+            method: spreheader::VERSION_PROPERTY.clone(),
+        }
+        .into();
+        assert!(try_eval_out_wo_ctx::<i8>(&expr).is_err());
+    }
+
+    // TODO: [sab] implement after merging https://github.com/ergoplatform/sigma-rust/pull/433
```
Comment (greenhat) [inline ergotree-interpreter/src/eval/spreheader.rs:170]: Let's uncomment it since #433 is merged.

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking great! Please uncomment the test for an unknown property.

File: ergotree-interpreter/src/eval/spreheader.rs (hunk 1 above)
Comment (SabaunT) [reply ergotree-interpreter/src/eval/spreheader.rs]: done
thank you for the review!

**Review by greenhat — APPROVED** [review]
Great! Thank you!

**Pull request merged**